Clang builds fail on macOS

**What version of protobuf and what language are you using?**
Version: master
Language: C++

**What operating system (Linux, Windows, ...) and version?**
macOS 10.14.5

**What runtime / compiler are you using (e.g., python version or gcc version)**

```
$ clang++ --version
> Apple LLVM version 10.0.1 (clang-1001.0.46.4)
> Target: x86_64-apple-darwin18.6.0
> Thread model: posix
> InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**What did you do?**

A simple build using cmake.

Steps to reproduce the behavior:

```
mkdir -p cmake/build
cd cmake/build
cmake .. -G Ninja
ninja
```

**What did you expect to see**

A successful build :)

**What did you see instead?**

```
[201/398] Linking CXX executable test_plugin
FAILED: test_plugin
: && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.12 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/usr/local/opt/zlib/lib -L/usr/local/opt/zlib/lib -L/usr/local/opt/zlib/lib CMakeFiles/test_plugin.dir/Users/benbader/Development/protobuf/src/google/protobuf/compiler/mock_code_generator.cc.o CMakeFiles/test_plugin.dir/Users/benbader/Development/protobuf/src/google/protobuf/testing/file.cc.o CMakeFiles/test_plugin.dir/Users/benbader/Development/protobuf/src/google/protobuf/compiler/test_plugin.cc.o  -o test_plugin  libprotoc.a libprotobuf.a libgmock.a /usr/lib/libz.dylib -latomic && :
ld: library not found for -latomic
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[202/398] Linking CXX executable protoc-3.9.0.1
FAILED: protoc-3.9.0.1
: && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.12 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/usr/local/opt/zlib/lib -L/usr/local/opt/zlib/lib -L/usr/local/opt/zlib/lib CMakeFiles/protoc.dir/Users/benbader/Development/protobuf/src/google/protobuf/compiler/main.cc.o  -o protoc-3.9.0.1  libprotoc.a libprotobuf.a /usr/lib/libz.dylib -latomic && :
ld: library not found for -latomic
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

This seems to have broken as of #6337; I'm able to build master prior to that PR landing.

**Anything else we should know about your project / environment**

I'm a little out of my depth in cmake, but it _looks_ as if the test for builtin atomics is incorrect.  When I looked in CMakeError.log, I see the following:

```
Performing C++ SOURCE FILE Test protobuf_HAVE_BUILTIN_ATOMICS failed with the following output:
Change Dir: /Users/benbader/Development/protobuf/cmake/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/local/bin/ninja cmTC_9cb94 
[1/2] Building CXX object CMakeFiles/cmTC_9cb94.dir/src.cxx.o
FAILED: CMakeFiles/cmTC_9cb94.dir/src.cxx.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++    -Dprotobuf_HAVE_BUILTIN_ATOMICS -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.12 -o CMakeFiles/cmTC_9cb94.dir/src.cxx.o -c src.cxx
src.cxx:4:34: error: expected '(' for function-style cast or type construction
      return std::atomic<int64_t>{};
             ~~~~~~~~~~~~~~~~~~~~^
1 error generated.
ninja: build stopped: subcommand failed.

Source file was:

    #include <atomic>
    int main() {
      return std::atomic<int64_t>{};
    }
  
```

The error is not about atomics, but about the initialization syntax.  I'm guessing that maybe the check needs to have --std=c++11 specified?

--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -137,6 +137,8 @@ endif (HAVE_ZLIB)
 set(protobuf_LINK_LIBATOMIC false)
 if (NOT MSVC)
   include(CheckCXXSourceCompiles)
+  set(OLD_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+  set(CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} -std=c++11)
   check_cxx_source_compiles("
     #include <atomic>
     int main() {
@@ -146,6 +148,7 @@ if (NOT MSVC)
   if (NOT protobuf_HAVE_BUILTIN_ATOMICS)
     set(protobuf_LINK_LIBATOMIC true)
   endif (NOT protobuf_HAVE_BUILTIN_ATOMICS)
+  set(CMAKE_REQUIRED_FLAGS ${OLD_CMAKE_REQUIRED_FLAGS})
 endif (NOT MSVC)
 
 if (protobuf_BUILD_SHARED_LIBS)

